File: src/main/java/org/railwaystations/rsapi/adapter/in/task/MastodonBotTask.java
package org.railwaystations.rsapi.adapter.in.task;

import lombok.extern.slf4j.Slf4j;
import org.railwaystations.rsapi.core.services.InboxService;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@Slf4j
public class MastodonBotTask {

    private final InboxService inboxService;

    public MastodonBotTask(InboxService inboxDao) {
        super();
        this.inboxService = inboxDao;
    }

    @Scheduled(fixedRate = 60_000 * 60) // every hour
    public void postNewPhoto() {
        log.info("Starting MastodonBotTask");
        inboxService.postRecentlyImportedPhotoNotYetPosted();
    }

}

File: service/src/main/kotlin/org/railwaystations/rsapi/adapter/task/MastodonBotTask.kt
package org.railwaystations.rsapi.adapter.task

import org.railwaystations.rsapi.core.ports.PostRecentlyImportedPhotoUseCase
import org.railwaystations.rsapi.core.utils.Logger
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class MastodonBotTask(private val postRecentlyImportedPhotoUseCase: PostRecentlyImportedPhotoUseCase) {

    private val log by Logger()

    @Scheduled(fixedRate = 60000 * 60) // every hour
    fun postNewPhoto() {
        log.info("Starting MastodonBotTask")
        postRecentlyImportedPhotoUseCase.postRecentlyImportedPhotoNotYetPosted()
    }
}
